### ISSUE TITLE
findReferences doesn't give results in the right order

### ISSUE THREAD

**Issue by JsonFreeman** [Microsoft/TypeScript#1853]
Search for "globalTemplateStringsArrayType" in checker.ts. The results are not in the right order (the order of the text spans).

**Comment by mhegazy** [comment]
@paulvanbrenk and @CyrusNajmabadi what is the ordering scheme we use here? do we even order them or just show the list that comes from the script side?

**Comment by CyrusNajmabadi** [comment]
I believe the ordering scheme is the ordering given back by the script side.  It is not manipulated at all.  The reasoning being that the individual language knows best what order to present stuff to the user.  And that order may not be in file order. For example, you may want to show the declaration first, then all the references.  But it's up to the language.

**Comment by DanielRosenwasser** [comment]
That's odd, I thought we were doing a full text search for candidates and deducing from there.

**Comment by steveluc** [comment]
I agree there should be some discretion by the editor, but there is a good reason for ordering the references.  If you process references by file, starting with the reference with the greatest position in the file and working your way to the top,  all of the references will still point to the right place even if you make edits in the ones further down in the file.

It looks odd to present references this way (inverse order), so editors often give the ability to pick the last one for a file and then to move to the previous one.

Steve
________________________________

**Comment by JsonFreeman** [comment]
I discovered the issue. The order of the references within a file is based on the row number and column number, but it is *lexicographic* in those numbers, and not numeric:
```ts
function foo() { } // ref #1
foo(); // ref #4
// line 3




// line 8
foo(); // ref #5
foo(); // ref #2
foo(); // ref #3
```

**Comment by JsonFreeman** [comment]
The implication of my comment is that it should be numeric, not lexicographic

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
